lazyflow/operators/classifierOperators.py

```diff
@@ -437,40 +438,15 @@ def execute(self, slot, subindex, roi, result):
             skip_prediction = not numpy.any(mask)
 
         if skip_prediction:
+            logger.debug(f"Skipping masked block {roi}")
             result[:] = 0.0
             return result
 
-        assert issubclass(type(classifier), LazyflowPixelwiseClassifierABC), \
-            "Classifier is of type {}, which does not satisfy the LazyflowPixelwiseClassifierABC interface."\
-            "".format( type(classifier) )
-
-        upstream_roi = (roi.start, roi.stop)
-        # Ask for the halo needed by the classifier
-        axiskeys = self.Image.meta.getAxisKeys()
-        halo_shape = classifier.get_halo_shape(axiskeys)
-        assert len(halo_shape) == len( upstream_roi[0] )
-        assert halo_shape[-1] == 0, "Didn't expect a non-zero halo for channel dimension."
-
-        # Expand block by halo, then clip to image bounds
-        upstream_roi = numpy.array( upstream_roi )
-        upstream_roi[0] -= halo_shape
-        upstream_roi[1] += halo_shape
-        upstream_roi = getIntersection( upstream_roi, roiFromShape(self.Image.meta.shape) )
-        upstream_roi = numpy.asarray( upstream_roi )
-
-        # Determine how to extract the data from the result (without the halo)
-        downstream_roi = numpy.array((roi.start, roi.stop))
-        predictions_roi = downstream_roi[:,:-1] - upstream_roi[0,:-1]
+        assert issubclass(type(classifier), self.classifierInterface), \
+            f"Classifier is of type {type(classifier)}, which does not satisfy the {self.classifierInterface} interface."
 
-        # Request all upstream channels
-        input_channels = self.Image.meta.shape[-1]
-        upstream_roi[:,-1] = [0, input_channels]
+        probabilities = self._calculate_probabilities(roi)
```

Comment:
I made the mehod abstract, but since the class inherits from operator, which already has a metaclass, I'm not sure I can safely make the class itself abstract =/

Reply:
You also can inherit from `abc.ABC` to get an abstract class. Without `abc` magic, `@abstractmethod` will do nothing in runtime, so in that case it's better to at least `raise NotImplementedError`.

Reply:
You also can inherit from `abc.ABC` to get an abstract class. Without `abc` magic, `@abstractmethod` will do nothing in runtime, so in that case it's better to at least `raise NotImplementedError`.

Reply:
Added inheritance do ABC. Maybe in the future we could move that into `Operator` itself

Reply:
Actually, `OperatorMetaClass` already inherits from `ABCMeta`, so you should be able to use `@abstractmethod` here without inheriting from `ABC`. Sorry for the confustion. Although it's strange that `Operator.execute` and friends are not defined as abstract methods.